Drop Python < 3.9 sys.version_info branches

gsnedders opened the pull request:
#### dcb4913d794e7f3bb07f3c4184abf912bcef2405
<pre>
Drop Python &lt; 3.9 sys.version_info branches
<a href="https://bugs.webkit.org/show_bug.cgi?id=274865">https://bugs.webkit.org/show_bug.cgi?id=274865</a>

Reviewed by Jonathan Bedard.

Largely based on ruff&apos;s fix for its outdated-version-block rule
(<a href="https://docs.astral.sh/ruff/rules/outdated-version-block/)">https://docs.astral.sh/ruff/rules/outdated-version-block/)</a>, with
`--target-version py39`, with some minor work before applying that
(fixing usage of sys.version doing string comparison, rewriting
sys.version_info.major to sys.version_info[0] as ruff doesn&apos;t support
the former).

With that done, the remaining usage of sys.version_info was audited
and, as needed, fixed manually.

Many of the prior comments about Python 2 v. 3 differences were also
addressed, with more code there becoming dead.

To tidy this up, usage of six was removed, now-unused imports were
removed, blank lines were removed, some dead assignments were removed,
and some usage of **kwargs was simplified where kwargs was now
constant.

We also now get rid of BeautifulSoup_legacy.

* Source/JavaScriptCore/KeywordLookupGenerator.py:
(Trie.coalesce):
* Source/JavaScriptCore/Scripts/cssmin.py:
* Source/JavaScriptCore/Scripts/generate-js-builtins.py:
(do_open):
* Source/JavaScriptCore/Scripts/jsmin.py:
(jsmin):
* Source/JavaScriptCore/Scripts/make-js-file-arrays.py:
(main):
* Source/JavaScriptCore/yarr/generateYarrCanonicalizeUnicode:
(openOrExit):
* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
(openOrExit):
* Source/WebCore/html/parser/create-html-entity-table:
* Source/WebKit/Scripts/webkit/model_unittest.py:
* Source/WebKit/Scripts/webkit/parser_unittest.py:
(UnsupportedPrecompilerDirectiveTest):
(UnsupportedPrecompilerDirectiveTest.assertRaisesRegex): Deleted.
* Tools/CISupport/download-built-product:
* Tools/CISupport/ews-app/ews/common/bugzilla.py:
(BugzillaBeautifulSoup._parse_attachment_ids_request_query):
* Tools/CISupport/ews-app/ews/thirdparty/BeautifulSoup.py:
(BeautifulSoup):
(BeautifulSoup.__init__):
(BeautifulStoneSoup):
(BeautifulStoneSoup.__init__):
* Tools/CISupport/ews-app/ews/thirdparty/BeautifulSoup_legacy.py: Removed.
* Tools/CISupport/kill-old-processes:
(are_simulators_booted):
* Tools/Scripts/dump-webkit-tests-run:
(main):
(convert_trie_to_flat_paths):
* Tools/Scripts/hooks/pre-commit:
* Tools/Scripts/hooks/pre-push:
* Tools/Scripts/hooks/prepare-commit-msg:
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla.request):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
(Package.archives):
(AutoInstall):
(AutoInstall.find_module):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/filtered_call.py:
(filtered_call):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/popen.py:
(Popen):
(Popen.__init__):
(Popen.communicate):
(Popen.wait):
(Popen.__enter__):
(Popen.__exit__):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/subprocess.py:
(Subprocess.completion_generator_for):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/terminal.py:
(Terminal.input):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/time_.py:
(Time):
(Time.__call__):
(add_metaclass): Deleted.
(add_metaclass.wrapper): Deleted.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/string_utils.py:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/subprocess_utils.py:
(run):
(TimeoutExpired): Deleted.
(TimeoutExpired.__init__): Deleted.
(TimeoutExpired.__str__): Deleted.
(TimeoutExpired.stdout): Deleted.
(CompletedProcess): Deleted.
(CompletedProcess.__init__): Deleted.
(CompletedProcess.__repr__): Deleted.
(CompletedProcess.check_returncode): Deleted.
(run.decode): Deleted.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/task_pool.py:
(TaskPool.__exit__):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py:
(Terminal.input):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/tests/subprocess_utils_unittest.py:
(SubprocessUtils.test_run_timeout):
(SubprocessUtils):
(SubprocessUtils.test_run_timeout_context):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/tests/terminal_unittest.py:
(TerminalTests.test_interrupt):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/timer.py:
* Tools/Scripts/libraries/webkitflaskpy/webkitflaskpy/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/commit_classifier.py:
(CommitClassifier.LineFilter.fuzzy):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git.Cache.populate):
(Git.__init__):
(Git.commits):
(Git.diff):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/svn.py:
(Svn.Cache.populate):
(Svn.commits):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.parser):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/command.py:
(FilteredCommand.pager):
(FilteredCommand.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_hooks.py:
(InstallHooks.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py:
(Pickable.Filters.fuzzy):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.fetch):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/scm_base.py:
(ScmBase.gmtoffset):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/hooks/prepare-commit-msg:
* Tools/Scripts/rebase-patch-after-webkit-move:
(needs_rebase):
(rebase_line):
* Tools/Scripts/sync-feature-defines:
* Tools/Scripts/validate-committer-lists:
* Tools/Scripts/webkitpy/__init__.py:
* Tools/Scripts/webkitpy/autoinstalled/buildbot.py:
* Tools/Scripts/webkitpy/autoinstalled/twisted.py:
* Tools/Scripts/webkitpy/benchmark_runner/benchmark_results_unittest.py:
(BenchmarkResultsTest):
(BenchmarkResultsTest.assertRaisesRegex): Deleted.
* Tools/Scripts/webkitpy/benchmark_runner/benchmark_runner.py:
(istext):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
(BrowserDriver):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/linux_browser_driver.py:
(LinuxBrowserDriver.launch_url):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_safari_driver.py:
(OSXSafariDriver._launch_url_with_custom_path):
* Tools/Scripts/webkitpy/benchmark_runner/http_server_driver/http_server_driver.py:
(HTTPServerDriver):
* Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
(WebServerBenchmarkRunner._construct_subtest_url):
* Tools/Scripts/webkitpy/common/checkout/checkout.py:
* Tools/Scripts/webkitpy/common/checkout/diff_parser.py:
(DiffParser._parse_into_diff_files):
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SCMTest.__init__):
(SCMTest.assertRegex): Deleted.
(SCMTest.assertNotRegex): Deleted.
* Tools/Scripts/webkitpy/common/iteration_compatibility.py:
(iteritems):
(iterkeys):
(itervalues):
* Tools/Scripts/webkitpy/common/message_pool.py:
* Tools/Scripts/webkitpy/common/net/bugzilla/bugzilla.py:
(Bugzilla._parse_attachment_element):
* Tools/Scripts/webkitpy/common/net/bugzilla/test_expectation_updater.py:
(argument_parser):
* Tools/Scripts/webkitpy/common/net/buildbot/buildbot.py:
* Tools/Scripts/webkitpy/common/net/credentials_unittest.py:
* Tools/Scripts/webkitpy/common/net/file_uploader.py:
* Tools/Scripts/webkitpy/common/net/networktransaction.py:
* Tools/Scripts/webkitpy/common/net/web.py:
* Tools/Scripts/webkitpy/common/net/web_mock.py:
* Tools/Scripts/webkitpy/common/system/executive.py:
(ScriptError.command_name):
(Executive.__init__):
(Executive._run_command_with_teed_output):
(Executive.run_command):
(Executive._child_process_encoding):
(Executive._should_encode_child_process_arguments):
(Executive.popen):
(WrappedPopen): Deleted.
(WrappedPopen.__init__): Deleted.
(WrappedPopen.returncode): Deleted.
(WrappedPopen.__enter__): Deleted.
(WrappedPopen.__exit__): Deleted.
(Executive._should_close_fds): Deleted.
* Tools/Scripts/webkitpy/common/system/filesystem_mock.py:
(ScanDirMock.__enter__):
(ScanDirMock.__exit__):
(ScanDirMock.close):
* Tools/Scripts/webkitpy/common/system/path.py:
* Tools/Scripts/webkitpy/common/system/profiler.py:
* Tools/Scripts/webkitpy/common/watchlist/watchlistparser_unittest.py:
(WatchListParserTest.test_bad_filename_regex):
(WatchListParserTest.test_bad_more_regex):
* Tools/Scripts/webkitpy/conftest.py:
(pytest_pycollect_makeitem):
* Tools/Scripts/webkitpy/featuredefines/search.py:
(FeatureDefinesSearch):
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder.py:
(LayoutTestFinder):
(LayoutTestFinder._percent_encoded_variant):
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder_legacy_unittest.py:
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests_integrationtest.py:
(RebaselineTest.test_reset_results):
(RebaselineTest.test_missing_results):
(RebaselineTest.test_new_baseline):
* Tools/Scripts/webkitpy/layout_tests/servers/web_platform_test_server.py:
(WebPlatformTestServer.__init__):
* Tools/Scripts/webkitpy/layout_tests/servers/websocket_server.py:
(PyWebSocket._prepare_config):
* Tools/Scripts/webkitpy/layout_tests/views/metered_stream_unittest.py:
(RegularTest.test_unicode):
(RegularTest.test_stream_with_encoding):
* Tools/Scripts/webkitpy/port/leakdetector_valgrind.py:
(LeakDetectorValgrind.parse_and_print_leaks_detail):
* Tools/Scripts/webkitpy/port/server_process_unittest.py:
(TestServerProcess):
* Tools/Scripts/webkitpy/port/xvfbdriver_unittest.py:
(XvfbDriverTest):
(XvfbDriverTest.assertRaisesRegex): Deleted.
* Tools/Scripts/webkitpy/results/upload.py:
(Upload.Encoder.default):
* Tools/Scripts/webkitpy/results/upload_unittest.py:
* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
(JSONImportExpectationsChecker.check):
* Tools/Scripts/webkitpy/style/checkers/jsonchecker_unittest.py:
(JSONCheckerTest.test_conflict_marker):
* Tools/Scripts/webkitpy/style/checkers/python.py:
(PythonChecker.check):
* Tools/Scripts/webkitpy/style/checkers/python_unittest.py:
(PythonCheckerTest.test_check):
* Tools/Scripts/webkitpy/style/main.py:
(CheckWebKitStyle):
(CheckWebKitStyle.main):
(CheckWebKitStyle._engage_awesome_stderr_hacks): Deleted.
* Tools/Scripts/webkitpy/test/finder.py:
(source_from_cache): Deleted.
* Tools/Scripts/webkitpy/test/main.py:
(main):
* Tools/Scripts/webkitpy/thirdparty/BeautifulSoup.py:
(BeautifulSoup):
(BeautifulSoup.__init__):
(BeautifulStoneSoup):
(BeautifulStoneSoup.__init__):
* Tools/Scripts/webkitpy/thirdparty/BeautifulSoup_legacy.py: Removed.
* Tools/Scripts/webkitpy/tool/commands/applywatchlistlocal_unittest.py:
(ApplyWatchListLocalTest):
(ApplyWatchListLocalTest.assertRaisesRegex): Deleted.
* Tools/Scripts/webkitpy/tool/servers/rebaselineserver.py:
* Tools/Scripts/webkitpy/tool/servers/reflectionhandler.py:
* Tools/Scripts/webkitpy/tool/steps/confirmdiff.py:
* Tools/Scripts/webkitpy/tool/steps/promptforbugortitle.py:
* Tools/Scripts/webkitpy/w3c/test_converter.py:
(_W3CTestConverter.__init__):
* Tools/Scripts/webkitpy/w3c/test_exporter.py:
* Tools/Scripts/webkitpy/w3c/wpt_github.py:
* Tools/glib/api_test_runner.py:
* Tools/glib/glib_test_runner.py:
(GLibTestRunner.run):
* Tools/lldb/lldb_webkit.py:
(bisect_right):
(btjs):

Canonical link: <a href="https://commits.webkit.org/287424@main">https://commits.webkit.org/287424@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f27e720fdaeeb5188461d712b4cb339a6c3b2b45

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/79393 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/58420 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/32767 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/83999 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/30524 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/81526 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/67481 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/6681 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/62100 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/19978 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/82460 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/52160 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/72334 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/42410 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/78904 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/49543 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/26520 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/28929 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/72539 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/70663 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/26983 "") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/85406 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/78577 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/6664 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/70347 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/79083 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/6829 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/68210 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/69594 "Found 1 new API test failure: /WebKitGTK/TestWebsiteData:/webkit/WebKitWebsiteData/service-worker-registrations (failure)") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/13640 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/12508 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/100924 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/12301 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/6616 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/12361 "Built successfully") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/24622 "Found 1 new JSC stress test failure: wasm.yaml/wasm/stress/struct-new_default-small-members.js.wasm-eager-jettison (failure)") | 
| | [❌ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/6535 "") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/10005 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/8332 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
```diff
@@ -460,8 +453,7 @@ class AutoInstall(object):
     BASE_LIBRARIES = ['setuptools', 'wheel', 'six', 'pyparsing', 'packaging', 'setuptools_scm']
-    if sys.version_info >= (3, 0):
-        BASE_LIBRARIES.insert(-1, 'tomli')
+    BASE_LIBRARIES.insert(-1, 'tomli')
```
We can stick this in the `BASE_LIBRARIES` list now, instead of inserting it.

JonWBedard commented on Tools/Scripts/webkitpy/common/message_pool.py:
```diff
@@ -48,12 +48,8 @@
-    import cPickle as pickle
-    import Queue as queue
+import pickle
+import queue
```
I need to see if we even use this class any more, I don't think we do.

gsnedders replied:
Yes, we do: https://github.com/WebKit/WebKit/blob/a01341c4156720f639df1eb1cfb8d3c153d54785/Tools/Scripts/webkitpy/common/message_pool.py#L76-L80

JonWBedard commented on Tools/Scripts/webkitpy/common/system/executive.py:
```diff
@@ -497,10 +497,7 @@ def popen(self, args, **kwargs):
-                    open_kwargs = {}
-                    if sys.version_info.major >= 3:
-                        open_kwargs['encoding'] = 'cp437'
+                    open_kwargs = {'encoding': 'cp437'}
```
We can feed these straight to `open` now.

JonWBedard commented on Tools/Scripts/hooks/pre-commit:
```diff
@@ -4,9 +4,7 @@ import os
-ENCODING_KWARGS = dict()
-if sys.version_info >= (3, 6):
-    ENCODING_KWARGS = dict(encoding='utf-8')
+ENCODING_KWARGS = dict(encoding='utf-8')
```
We should probably just feed this straight into the `subprocess` call now, although this is probably better as a follow-up fix.

gsnedders commented on Tools/Scripts/webkitpy/thirdparty/wpt/manifest/XMLParser.py:
```diff
@@ -39,11 +39,7 @@ def _fixname(key):
-else:
-    _codes = {expat.ErrorString(i): i for i in range(0x100)}  # type: Dict[str, int]
-    _undefined_entity_code = _codes[expat.errors.XML_ERROR_UNDEFINED_ENTITY]
+_undefined_entity_code = expat.errors.codes[expat.errors.XML_ERROR_UNDEFINED_ENTITY]
```
This shouldn't have gotten changed, given this is imported code.

gsnedders committed:
Drop Python < 3.9 sys.version_info branches
https://bugs.webkit.org/show_bug.cgi?id=274865

Reviewed by Jonathan Bedard.

Largely based on ruff's fix for its outdated-version-block rule
(https://docs.astral.sh/ruff/rules/outdated-version-block/), with
`--target-version py39`, with some minor work before applying that
(fixing usage of sys.version doing string comparison, rewriting
sys.version_info.major to sys.version_info[0] as ruff doesn't support
the former).

With that done, the remaining usage of sys.version_info was audited
and, as needed, fixed manually.

Many of the prior comments about Python 2 v. 3 differences were also
addressed, with more code there becoming dead.

To tidy this up, usage of six was removed, now-unused imports were
removed, blank lines were removed, some dead assignments were removed,
and some usage of **kwargs was simplified where kwargs was now
constant.

We also now get rid of BeautifulSoup_legacy.

* Source/JavaScriptCore/KeywordLookupGenerator.py:
(Trie.coalesce):
* Source/JavaScriptCore/Scripts/cssmin.py:
* Source/JavaScriptCore/Scripts/generate-js-builtins.py:
(do_open):
* Source/JavaScriptCore/Scripts/jsmin.py:
(jsmin):
* Source/JavaScriptCore/Scripts/make-js-file-arrays.py:
(main):
* Source/JavaScriptCore/yarr/generateYarrCanonicalizeUnicode:
(openOrExit):
* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
(openOrExit):
* Source/WebCore/html/parser/create-html-entity-table:
* Source/WebKit/Scripts/webkit/model_unittest.py:
* Source/WebKit/Scripts/webkit/parser_unittest.py:
(UnsupportedPrecompilerDirectiveTest):
(UnsupportedPrecompilerDirectiveTest.assertRaisesRegex): Deleted.
* Tools/CISupport/download-built-product:
* Tools/CISupport/ews-app/ews/common/bugzilla.py:
(BugzillaBeautifulSoup._parse_attachment_ids_request_query):
* Tools/CISupport/ews-app/ews/thirdparty/BeautifulSoup.py:
(BeautifulSoup):
(BeautifulSoup.__init__):
(BeautifulStoneSoup):
(BeautifulStoneSoup.__init__):
* Tools/CISupport/ews-app/ews/thirdparty/BeautifulSoup_legacy.py: Removed.
* Tools/CISupport/kill-old-processes:
(are_simulators_booted):
* Tools/Scripts/dump-webkit-tests-run:
(main):
(convert_trie_to_flat_paths):
* Tools/Scripts/hooks/pre-commit:
* Tools/Scripts/hooks/pre-push:
* Tools/Scripts/hooks/prepare-commit-msg:
* Tools/Scripts/libraries/resultsdbpy/resultsdbpy/__init__.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla.request):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/__init__.py:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/autoinstall.py:
(Package.archives):
(AutoInstall):
(AutoInstall.find_module):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/filtered_call.py:
(filtered_call):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/popen.py:
(Popen):
(Popen.__init__):
(Popen.communicate):
(Popen.wait):
(Popen.__enter__):
(Popen.__exit__):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/subprocess.py:
(Subprocess.completion_generator_for):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/terminal.py:
(Terminal.input):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/mocks/time_.py:
(Time):
(Time.__call__):
(add_metaclass): Deleted.
(add_metaclass.wrapper): Deleted.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/string_utils.py:
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/subprocess_utils.py:
(run):
(TimeoutExpired): Deleted.
(TimeoutExpired.__init__): Deleted.
(TimeoutExpired.__str__): Deleted.
(TimeoutExpired.stdout): Deleted.
(CompletedProcess): Deleted.
(CompletedProcess.__init__): Deleted.
(CompletedProcess.__repr__): Deleted.
(CompletedProcess.check_returncode): Deleted.
(run.decode): Deleted.
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/task_pool.py:
(TaskPool.__exit__):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/terminal.py:
(Terminal.input):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/tests/subprocess_utils_unittest.py:
(SubprocessUtils.test_run_timeout):
(SubprocessUtils):
(SubprocessUtils.test_run_timeout_context):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/tests/terminal_unittest.py:
(TerminalTests.test_interrupt):
* Tools/Scripts/libraries/webkitcorepy/webkitcorepy/timer.py:
* Tools/Scripts/libraries/webkitflaskpy/webkitflaskpy/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/commit_classifier.py:
(CommitClassifier.LineFilter.fuzzy):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/git.py:
(Git.Cache.populate):
(Git.__init__):
(Git.commits):
(Git.diff):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/local/svn.py:
(Svn.Cache.populate):
(Svn.commits):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/__init__.py:
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
(Branch.parser):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/command.py:
(FilteredCommand.pager):
(FilteredCommand.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/install_hooks.py:
(InstallHooks.main):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pickable.py:
(Pickable.Filters.fuzzy):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/setup.py:
(Setup.fetch):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/scm_base.py:
(ScmBase.gmtoffset):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/hooks/prepare-commit-msg:
* Tools/Scripts/rebase-patch-after-webkit-move:
(needs_rebase):
(rebase_line):
* Tools/Scripts/sync-feature-defines:
* Tools/Scripts/validate-committer-lists:
* Tools/Scripts/webkitpy/__init__.py:
* Tools/Scripts/webkitpy/autoinstalled/buildbot.py:
* Tools/Scripts/webkitpy/autoinstalled/twisted.py:
* Tools/Scripts/webkitpy/benchmark_runner/benchmark_results_unittest.py:
(BenchmarkResultsTest):
(BenchmarkResultsTest.assertRaisesRegex): Deleted.
* Tools/Scripts/webkitpy/benchmark_runner/benchmark_runner.py:
(istext):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/browser_driver.py:
(BrowserDriver):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/linux_browser_driver.py:
(LinuxBrowserDriver.launch_url):
* Tools/Scripts/webkitpy/benchmark_runner/browser_driver/osx_safari_driver.py:
(OSXSafariDriver._launch_url_with_custom_path):
* Tools/Scripts/webkitpy/benchmark_runner/http_server_driver/http_server_driver.py:
(HTTPServerDriver):
* Tools/Scripts/webkitpy/benchmark_runner/webserver_benchmark_runner.py:
(WebServerBenchmarkRunner._construct_subtest_url):
* Tools/Scripts/webkitpy/common/checkout/checkout.py:
* Tools/Scripts/webkitpy/common/checkout/diff_parser.py:
(DiffParser._parse_into_diff_files):
* Tools/Scripts/webkitpy/common/checkout/scm/scm_unittest.py:
(SCMTest.__init__):
(SCMTest.assertRegex): Deleted.
(SCMTest.assertNotRegex): Deleted.
* Tools/Scripts/webkitpy/common/iteration_compatibility.py:
(iteritems):
(iterkeys):
(itervalues):
* Tools/Scripts/webkitpy/common/message_pool.py:
* Tools/Scripts/webkitpy/common/net/bugzilla/bugzilla.py:
(Bugzilla._parse_attachment_element):
* Tools/Scripts/webkitpy/common/net/bugzilla/test_expectation_updater.py:
(argument_parser):
* Tools/Scripts/webkitpy/common/net/buildbot/buildbot.py:
* Tools/Scripts/webkitpy/common/net/credentials_unittest.py:
* Tools/Scripts/webkitpy/common/net/file_uploader.py:
* Tools/Scripts/webkitpy/common/net/networktransaction.py:
* Tools/Scripts/webkitpy/common/net/web.py:
* Tools/Scripts/webkitpy/common/net/web_mock.py:
* Tools/Scripts/webkitpy/common/system/executive.py:
(ScriptError.command_name):
(Executive.__init__):
(Executive._run_command_with_teed_output):
(Executive.run_command):
(Executive._child_process_encoding):
(Executive._should_encode_child_process_arguments):
(Executive.popen):
(WrappedPopen): Deleted.
(WrappedPopen.__init__): Deleted.
(WrappedPopen.returncode): Deleted.
(WrappedPopen.__enter__): Deleted.
(WrappedPopen.__exit__): Deleted.
(Executive._should_close_fds): Deleted.
* Tools/Scripts/webkitpy/common/system/filesystem_mock.py:
(ScanDirMock.__enter__):
(ScanDirMock.__exit__):
(ScanDirMock.close):
* Tools/Scripts/webkitpy/common/system/path.py:
* Tools/Scripts/webkitpy/common/system/profiler.py:
* Tools/Scripts/webkitpy/common/watchlist/watchlistparser_unittest.py:
(WatchListParserTest.test_bad_filename_regex):
(WatchListParserTest.test_bad_more_regex):
* Tools/Scripts/webkitpy/conftest.py:
(pytest_pycollect_makeitem):
* Tools/Scripts/webkitpy/featuredefines/search.py:
(FeatureDefinesSearch):
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder.py:
(LayoutTestFinder):
(LayoutTestFinder._percent_encoded_variant):
* Tools/Scripts/webkitpy/layout_tests/controllers/layout_test_finder_legacy_unittest.py:
* Tools/Scripts/webkitpy/layout_tests/run_webkit_tests_integrationtest.py:
(RebaselineTest.test_reset_results):
(RebaselineTest.test_missing_results):
(RebaselineTest.test_new_baseline):
* Tools/Scripts/webkitpy/layout_tests/servers/web_platform_test_server.py:
(WebPlatformTestServer.__init__):
* Tools/Scripts/webkitpy/layout_tests/servers/websocket_server.py:
(PyWebSocket._prepare_config):
* Tools/Scripts/webkitpy/layout_tests/views/metered_stream_unittest.py:
(RegularTest.test_unicode):
(RegularTest.test_stream_with_encoding):
* Tools/Scripts/webkitpy/port/leakdetector_valgrind.py:
(LeakDetectorValgrind.parse_and_print_leaks_detail):
* Tools/Scripts/webkitpy/port/server_process_unittest.py:
(TestServerProcess):
* Tools/Scripts/webkitpy/port/xvfbdriver_unittest.py:
(XvfbDriverTest):
(XvfbDriverTest.assertRaisesRegex): Deleted.
* Tools/Scripts/webkitpy/results/upload.py:
(Upload.Encoder.default):
* Tools/Scripts/webkitpy/results/upload_unittest.py:
* Tools/Scripts/webkitpy/style/checkers/jsonchecker.py:
(JSONImportExpectationsChecker.check):
* Tools/Scripts/webkitpy/style/checkers/jsonchecker_unittest.py:
(JSONCheckerTest.test_conflict_marker):
* Tools/Scripts/webkitpy/style/checkers/python.py:
(PythonChecker.check):
* Tools/Scripts/webkitpy/style/checkers/python_unittest.py:
(PythonCheckerTest.test_check):
* Tools/Scripts/webkitpy/style/main.py:
(CheckWebKitStyle):
(CheckWebKitStyle.main):
(CheckWebKitStyle._engage_awesome_stderr_hacks): Deleted.
* Tools/Scripts/webkitpy/test/finder.py:
(source_from_cache): Deleted.
* Tools/Scripts/webkitpy/test/main.py:
(main):
* Tools/Scripts/webkitpy/thirdparty/BeautifulSoup.py:
(BeautifulSoup):
(BeautifulSoup.__init__):
(BeautifulStoneSoup):
(BeautifulStoneSoup.__init__):
* Tools/Scripts/webkitpy/thirdparty/BeautifulSoup_legacy.py: Removed.
* Tools/Scripts/webkitpy/tool/commands/applywatchlistlocal_unittest.py:
(ApplyWatchListLocalTest):
(ApplyWatchListLocalTest.assertRaisesRegex): Deleted.
* Tools/Scripts/webkitpy/tool/servers/rebaselineserver.py:
* Tools/Scripts/webkitpy/tool/servers/reflectionhandler.py:
* Tools/Scripts/webkitpy/tool/steps/confirmdiff.py:
* Tools/Scripts/webkitpy/tool/steps/promptforbugortitle.py:
* Tools/Scripts/webkitpy/w3c/test_converter.py:
(_W3CTestConverter.__init__):
* Tools/Scripts/webkitpy/w3c/test_exporter.py:
* Tools/Scripts/webkitpy/w3c/wpt_github.py:
* Tools/glib/api_test_runner.py:
* Tools/glib/glib_test_runner.py:
(GLibTestRunner.run):
* Tools/lldb/lldb_webkit.py:
(bisect_right):
(btjs):

Canonical link: https://commits.webkit.org/287424@main
